### PR TITLE
MPDX-6943-Add-Newsletter-Dropdown-Functionality

### PR DESCRIPTION
**Pull request by TheNoodleMoose**
So this ticket adds functionality to the newsletter menu, but to a certain degree. This menu has three dropdown options:
1. Log Newsletter
2. Export Email
3. Export Physical

Only "Export Email" has full functionality and design in this PR, and I plan to create separate tickets/prs to address the other two. I did this as I didn't want the pr to get too large. "Log Newsletter" and "Export Physical" are just the barebones for those dialogs and tests.  Let me know what you think and if you have any questions 🙂

### CODE REVIEW

File: pages/api/graphql-rest.page.ts
Comment (TheNoodleMoose) [inline pages/api/graphql-rest.page.ts:30]: Was super happy to figure this out. I created a custom scalar here and then use it up in my `TaskAnalytics` type. On codegen, https://github.com/CruGlobal/mpdx-react/blob/7554b164b815d4fc08d60695e13d2c7cdea68524/codegen-server.yml#L17-L18 sees this scalar and will generate the correct type for it.

File: pages/api/graphql-rest.page.ts
Comment (TheNoodleMoose) [inline pages/api/graphql-rest.page.ts:64]: Once this is merged in, I'd also like to take a crack at trying to modularize this and break these queries and types up into different files.

File: src/components/Dashboard/ThisWeek/NewsletterMenu/NewsletterMenu.stories.tsx
Comment (TheNoodleMoose) [inline src/components/Dashboard/ThisWeek/NewsletterMenu/NewsletterMenu.stories.tsx:20]: For some reason, ergonomock was always returning null, even though the type should be a ```string```. I think it may have something to do with it being a custom scalar, but not too sure. To fix it, I just had to mock the specific dates I wanted.

File: src/components/Dashboard/ThisWeek/NewsletterMenu/NewsletterMenu.test.tsx
Comment (TheNoodleMoose) [inline src/components/Dashboard/ThisWeek/NewsletterMenu/NewsletterMenu.test.tsx:45]: As I mentioned, I created super barebones tests for the two menu options that I didn't fully fledge out in this pr.

File: src/components/Dashboard/ThisWeek/NewsletterMenu/NewsletterMenu.test.tsx
Comment (TheNoodleMoose) [inline src/components/Dashboard/ThisWeek/NewsletterMenu/NewsletterMenu.test.tsx:115]: I had to do some googling to figure out how to test copying text to the clipboard. This is what I based this on: https://stackoverflow.com/questions/62351935/how-to-mock-navigator-clipboard-writetext-in-jest

File: src/components/Dashboard/ThisWeek/NewsletterMenu/NewsletterMenu.tsx
Comment (TheNoodleMoose) [inline src/components/Dashboard/ThisWeek/NewsletterMenu/NewsletterMenu.tsx:30]: I really prefer the ```styled``` API over what we were doing before. I may be biased though as it's very similar to what I always use in https://styled-components.com/ or https://emotion.sh/docs/@emotion/styled 🙂

File: src/components/Dashboard/ThisWeek/NewsletterMenu/NewsletterMenu.tsx
Comment (TheNoodleMoose) [inline src/components/Dashboard/ThisWeek/NewsletterMenu/NewsletterMenu.tsx:153]: Also very important feedback I'd love to hear, how do you feel about how I approached rendering these dialogs? Could it be done better? I didn't want to create a separate component for each of these menu items so this seemed like the best way to handle it. Let me know what you guys think 🙂

File: pages/api/graphql-rest.page.ts
Comment (frett) [reply pages/api/graphql-rest.page.ts:64]: just a heads up that I've been modifying this file as well to tweak the data that is returned for contact filters

Comment (TheNoodleMoose) [reply pages/api/graphql-rest.page.ts:64]: Okay, thanks for the heads up. I'll wait then until it's in a good place to be modified.

File: src/components/Dashboard/ThisWeek/NewsletterMenu/NewsletterMenu.tsx
Comment (j2trumpet) [inline src/components/Dashboard/ThisWeek/NewsletterMenu/NewsletterMenu.tsx:34]: @TheNoodleMoose I'd change this gray tone to `theme.palette.text.primary`. I'm not sure where it's defined, but the light gray tone used for the background circle on hover should be `#EBECEC`. That light gray shade isn't available in the theme yet, but if you hard code it, I'll fix it as I'm working on the theme overhaul.

File: src/components/Dashboard/ThisWeek/NewsletterMenu/NewsletterMenu.tsx
Comment (j2trumpet) [inline src/components/Dashboard/ThisWeek/NewsletterMenu/NewsletterMenu.tsx:40]: @TheNoodleMoose I don't think you need these three style rules. Turning them off in local testing and adding `textAlign: 'left'` achieves the same look.

File: src/components/Dashboard/ThisWeek/NewsletterMenu/NewsletterMenu.stories.tsx
Comment (gyasistory) [inline src/components/Dashboard/ThisWeek/NewsletterMenu/NewsletterMenu.stories.tsx:1]:
```suggestion
import { MockedProvider } from '@apollo/client/testing';
```

**Review by gyasistory — CHANGES_REQUESTED** [review]

File: src/components/Dashboard/ThisWeek/NewsletterMenu/NewsletterMenu.test.tsx
Comment (TheNoodleMoose) [inline src/components/Dashboard/ThisWeek/NewsletterMenu/NewsletterMenu.test.tsx:230]: I added this to show that the logic will filter out a value from the email list string if the person does not have an ```email```.

**Comment by TheNoodleMoose** [comment]
I've decided to go ahead and move the different menu items into a new folder and be contained to their own components.

File: src/components/Dashboard/ThisWeek/NewsletterMenu/NewsletterMenu.tsx
Comment (OzzieOrca) [inline src/components/Dashboard/ThisWeek/NewsletterMenu/NewsletterMenu.tsx:82]: Use `useMemo` for this instead of `useState` and `useEffect`.

File: src/components/Dashboard/ThisWeek/NewsletterMenu/NewsletterMenu.tsx
Comment (OzzieOrca) [inline src/components/Dashboard/ThisWeek/NewsletterMenu/NewsletterMenu.tsx:85]: a `.map` to get an array of emails and a `.join(',')` might be cleaner than the reduce you have. Might have to throw a `.filter(Boolean)` in there if you need null checks. 🤷‍♂️

File: src/components/Dashboard/ThisWeek/NewsletterMenu/NewsletterMenu.tsx
Comment (OzzieOrca) [inline src/components/Dashboard/ThisWeek/NewsletterMenu/NewsletterMenu.tsx:56]: I think using undefined is cleaner `useState<SomeNonNullType>()`. I've been cleaning up some of these cuz TS strict wants a type here.

File: src/components/Dashboard/ThisWeek/NewsletterMenu/NewsletterMenu.tsx
Comment (OzzieOrca) [reply src/components/Dashboard/ThisWeek/NewsletterMenu/NewsletterMenu.tsx:154]: Instead of several different switches, might be better to group the title, content, action etc into an array of objects... 🤷‍♂️

File: src/components/Dashboard/ThisWeek/NewsletterMenu/NewsletterMenu.tsx
Comment (TheNoodleMoose) [reply src/components/Dashboard/ThisWeek/NewsletterMenu/NewsletterMenu.tsx:85]: Yep, I like that better 🙂

File: src/components/Dashboard/ThisWeek/NewsletterMenu/NewsletterMenu.tsx
Comment (TheNoodleMoose) [reply src/components/Dashboard/ThisWeek/NewsletterMenu/NewsletterMenu.tsx:82]: Good idea, been a while since I've used ```usedMemo```!

File: src/components/Dashboard/ThisWeek/NewsletterMenu/NewsletterMenu.tsx
Comment (OzzieOrca) [inline src/components/Dashboard/ThisWeek/NewsletterMenu/NewsletterMenu.tsx:123]: This seems bad...

**Review by OzzieOrca — APPROVED** [review]

File: src/components/Dashboard/ThisWeek/NewsletterMenu/NewsletterMenu.tsx
Comment (TheNoodleMoose) [reply src/components/Dashboard/ThisWeek/NewsletterMenu/NewsletterMenu.tsx]: Oh I missed that, thanks 👍

**Review by OzzieOrca — APPROVED** [review]

**Comment by TheNoodleMoose** [comment]
@OzzieOrca Looks like there was a type change on the API. Should I merge this in and we can address the type issue in a separate pr or should I wait until that fix is merged in? The type is unrelated to my work here.

**Comment by OzzieOrca** [comment]
It's unrelated, let's just fix it in a new PR.

**Pull request merged**
